### ISSUE TITLE
How to build a helm chart to push to a container registry?

### ISSUE THREAD

**Issue by streamnsight** [helm/helm#12013]
Hello,

I have a use case where I need to push my helm chart to a container registry. 

It is easily done by creating the tar.gz file and using `helm push <tar-file> repo-url` however, I have a private registry and this needs to be done via a resource manager on a cloud provider where I can get the registry login access_token, but i don't have access to user creds.

`docker login` requires the user / password to get the access_token for the registry, and then authenticate there. I don't have these creds, I have the token for the registry directly. I can use the registry HTTP API.

`helm registry login` is the same way, it requires the user / password and go fetch the auth token to then authenticate with the registry. So I cannot use `helm registry login` or `docker login` to login, so I can't use `helm push` directly.

In short, I can only use the registry HTTP API, and that means I need to be able to build the chart layers to upload to the registry directly.
I understand the chart is packaged as a couple binary layers in the container registry: how do I build those layers with helm or some other tool?

I tried pushing a chart with helm from local (with user creds) and then pull the container image from the registry with `docker pul` to see what it looks like, but docker pull errors saying there is an `invalid rootfs in image configuration`. 
It kinda makes sense as it is not a valid container image.

Where can I find docs on how to packages these layers to upload with HTTP API?

Thanks
